src/createServer.js

```diff
@@ -1,10 +1,66 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    if (req.url !== '/add-expense') {
+      res.statusCode = 404;
+
+      return res.end('Wrong request URL');
+    }
+
+    const chunks = [];
+
+    req.on('data', (chunk) => chunks.push(chunk));
+
+    req.on('end', () => {
+      const text = Buffer.concat(chunks).toString();
+      let data;
+
+      try {
+        data = JSON.parse(text);
```

Comment:
maybe you could've let more logic into this try here (like the if with the data) but in a sense it's better to keep separate try's if there's more logic that could throw errors, anyways, nice!